Mathlib/Algebra/Group/Irreducible/Indecomposable.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2025 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+module
+
+public import Mathlib.Algebra.Group.Irreducible.Defs
+public import Mathlib.Algebra.Group.Submonoid.Basic
+public import Mathlib.Algebra.Order.Monoid.Defs
+public import Mathlib.Order.Preorder.Finite
+
+/-!
+# Indecomposable elements of monoids
+-/
+
+@[expose] public section
+
+open Set
+
+variable {ι M : Type*} [Monoid M]
+
+/-- Given a family of elements of a monoid, a member is said to be indecomposable if it cannot be
+written as a product of two others in a non-trivial way. -/
+@[to_additive (attr := simp) /-- Given a family of elements of an additive monoid, a member is said
+to be indecomposable if it cannot be written as a sum of two others in a non-trivial way.-/]
+def IsMulIndecomposable (v : ι → M) (s : Set ι) (i : ι) : Prop :=
+  i ∈ s ∧ ∀ᵉ (j ∈ s) (k ∈ s), v i = v j * v k → v j = 1 ∨ v k = 1
+
+@[to_additive]
+lemma isMulIndecomposable_id_univ [Subsingleton Mˣ] {x : M} (hx : x ≠ 1) :
```

Comment:
Hmm, do you expect to use this lemma in the future?
I was rather expecting a connection to eg linear independence.
Anyways, not a blocker for me.